docs/pages.jl

```diff
@@ -1,24 +1,69 @@
-                                            "catalyst_functionality/constraint_equations.md",
-                                            "catalyst_functionality/parametric_stoichiometry.md",
-                                            "catalyst_functionality/network_analysis.md",
-                                            "catalyst_functionality/chemistry_related_functionality.md",
-                                            "Model creation examples" => Any["catalyst_functionality/example_networks/basic_CRN_examples.md",
-                                                                             "catalyst_functionality/example_networks/hodgkin_huxley_equation.md",
-                                                                             "catalyst_functionality/example_networks/smoluchowski_coagulation_equation.md"]],
-            "Catalyst Applications" => Any["catalyst_applications/simulation_structure_interfacing.md",
-                                           "catalyst_applications/advanced_simulations.md",
-                                           "catalyst_applications/homotopy_continuation.md",
-                                           "catalyst_applications/nonlinear_solve.md",
-                                           "catalyst_applications/bifurcation_diagrams.md"],
-            "Inverse Problems" => Any["inverse_problems/optimization_ode_param_fitting.md",
-                                        #"inverse_problems/petab_ode_param_fitting.md",
-                                        "inverse_problems/structural_identifiability.md",
-                                        "Inverse problem examples" => Any["inverse_problems/examples/ode_fitting_oscillation.md"]],
-            "FAQs" => "faqs.md",
-            "API" => "api/catalyst_api.md"]
\ No newline at end of file
+pages = Any[
+    "Home" => "home.md",
+    "Introduction to Catalyst" => Any[
+        "01_introduction_to_catalyst/01_catalyst_for_new_julia_users.md",
+        "01_introduction_to_catalyst/02_introduction_to_catalyst.md"
+        # Advanced introduction.
+    ],
+    "Model Creation and Properties" => Any[
+        "02_model_creation/01_dsl_description.md",
+        # DSL - Advanced
+        "02_model_creation/03_programmatic_CRN_construction.md",
+        "02_model_creation/04_constraint_equations.md",
+        "02_model_creation/05_compositional_modeling.md",
+        # Events.
+        # Distributed parameters, rates, and initial conditions.
+        # Loading and writing models to files.
+        # Model visualisation.
+        "02_model_creation/10_network_analysis.md",
+        "02_model_creation/11_chemistry_related_functionality.md",
+        "Model creation examples" => Any[
+            "02_model_creation/examples/basic_CRN_examples.md",
+            "02_model_creation/examples/hodgkin_huxley_equation.md",
+            "02_model_creation/examples/smoluchowski_coagulation_equation.md"
+        ]
+    ],
+    "Model simulation" => Any[
```

Comment:
Maybe calls this “Simulating Dynamics” to contrast with Steady-State analysis and make clear this is about temporal simulation?

Reply:
Kind of makes sense, just feel "Model simulation" is very clear what it means, and it is something basic enough that it is not too surprising that it occur in different settings (e.g. spatial simulations)

Reply:
But calculating steady-states is also model simulation so it is weird to put it in a different category…

Reply:
If you want to move it into “model simulation” as a sub-folder then that would be another way to address this.

Reply:
But model simulation is part of most other categories as well? E.g. the spatial modelling is basically just specific types of simulations.

Given that we have a specific category for finding steady states, it makes sense to have the simulation-based approach for that there as opposed to separating it from e.g. HC and nonlinear-based approaches.

If you have a better name for this section I am happy to change it, but “Simulating Dynamics” sounds weirdly convoluted to me.

Reply:
> But model simulation is part of most other categories as well? E.g. the spatial modelling is basically just specific types of simulations.

that’s exactly my point. It is strange to have that simulating the dynamics of ODE, SDE, and jump models is “model simulation”, but none of the other stuff is. Hence why I suggest changing the name of that section or moving stuff that is also model simulation into it as sub-sections.

alternatively you could split model simulation into an ODE model section and a stochastic model section, with the former including SS analysis too.

Reply:
What is your suggestion though? Creating separate simulation sections doesn't really solve it. As I mentioned, I really do not see how steady state simulations belong in any other category but the one which includes HC and nonlinear approaches for finding steady states.

I don't really see any problem with calling the current section "Model Simulation", but if you have a different name that is also good I do not mind changing.

Reply:
I’m not suggesting splitting up the SS category. I’m suggesting moving everything from that category together into model simulation so it is a nested sub-category.

Reply:
You could also then call it well-mixed model simulation to distinguish from spatial model simulation.

Reply:
Right, but then we have Bifurcation analysis and finding steady states through homotopy continuation as a sub-category to model simulations? Does that really make sense?

I don't see any reason we would want to have a different categorisation than the current one. I am happy to hear arguments, but I don't think that is what we are discussing. I.e. there is one discussion on how to categorise tutorials in groups with related stuff, and then there is the question of what names for the various tutorials and group makes sense. If I understand it right we are discussing the second part?

Regarding the names, I think it makes sense that you have a "Simulation" category which covers how simulation works. Since CRN modelling is pretty much about simulations, it is natural that other categories involves it (i.e. steady state simulations, spatial simulations, simulations to solve the inverse problem). Then stuff which is discussed in the simulation section is naturally relevant to these. In this sense I think it makes sense to have a simulation section, and it is not weird that spatial simulations are covered in the spatial modelling section (which also includes how to create spatial models, which is distinct from the model creation sections).

If we want to be precise we would probably have to call it "Forward non-steady state simulations of well-mixed systems" (or towards that), but that would only be weird. I don't see how "Simulation Dynamics" really make anything better or clearer. "Simulation Basics" could work, but I wouldn't really can stuff in there basics.

Reply:
If you think the current categorisation is non-intuitive, I am happy to discuss alternatives. If not, I think we should move toward discussing category names (and if we cannot come up with good names within the next day, we should just use some alternative, raise an issue, and continue the discussion in parallel as we work towards v14).

Reply:
> Right, but then we have Bifurcation analysis and finding steady states through homotopy continuation as a sub-category to model simulations? Does that really make sense?

Yes it does. 

Perhaps the organization should be focused on the mathematical model one is working with instead. In that case a splitting like ODE Models (which includes SS analysis as a sub-category as it is currently only about ODE models), and Stochastic Models makes more sense. The latter can include SDE/jumps along with ensemble problems and stats calculations. We can potentially split Stochastic Models in the future if there is a enough material to warrant separate jump/SDE categories.

The benefit to this is a I suspect most users are just interested in ODE models, and so everything they might want to do their with regard to solving such models would be collected together sequentially.

Reply:
I def. think there is a point, however, I do not really agree. Stuff like simulation structure interfacing and plotting are common to all categories. Looking at the ways to do it, I think that I have proposed here feels most intuitive.

Practically, whenever I try to work around what you are suggesting to an actual structure, things become weird. How do you actually propose to sort the new docs?